### PR TITLE
Expose UDF in a view to retrieve workfiles used diskspace

### PR DESCRIPTION
**Pull request by hardikar**
[#130666969]
Internally this is the shared memory variable used_segspace in
the view gp_toolkit.gp_workfile_mgr_used_diskspace.

Signed-off-by: Xin Zhang <xzhang@pivotal.io>

@karthijrk @armenatzoglou

### CODE REVIEW

**Comment by karthijrk** [comment]
LGTM 👍

**Comment by armenatzoglou** [comment]
+1

File: src/backend/catalog/gp_toolkit.sql [hunk 1]
```diff
@@ -2135,6 +2136,55 @@ GRANT SELECT ON gp_toolkit.gp_workfile_usage_per_query TO public;
...
+--        gp_toolkit.__gp_workfile_mgr_used_diskspace_f
+--
+-- @in:
+--
+-- @out:
+--        int - segment id
+--        bigint - size in bytes,
+--
+-- @doc:
+--        UDF to retrieve workfile used diskpace counter in bytes per segment
```
Comment (danielgustafsson) [inline src/backend/catalog/gp_toolkit.sql]: s/diskpace/diskspace/

**Comment by asimrp** [comment]
@deepesh1081 do you know if we run gp_toolkit.sql for minor version upgrades in 4.3?

**Comment by ivannovick** [comment]
Minor versions don't run ANY utilities its a binary swap only.  In the past we have provided new .sql files for users as optional step only during upgrade for updated views.

File: src/backend/catalog/gp_toolkit.sql (hunk 1 above)
Comment (xinzweb) [inline src/backend/catalog/gp_toolkit.sql]: Nice catch. Thanks.

**Pull request merged**